analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTests.cs

```diff
@@ -0,0 +1,160 @@
+            sut.OutPath.Should().Be(@"C:\foo\bar\.sonarqube\out\0");
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().Be("netcoreapp3.1");
+        }
+
+        [TestMethod]
+        public void WhenHasUnexpectedProjectType_FallsBackToProduct()
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\UnexpectedProjectTypeValue\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.ProjectType.Should().Be(ProjectType.Product);
+        }
+
+        [DataTestMethod]
+        [DataRow("MissingAnalysisConfigPath")]
+        [DataRow("MissingOutPath")]
+        [DataRow("MissingProjectPath")]
+        [DataRow("MissingProjectType")]
+        [DataRow("MissingTargetFramework")]
+        public void WhenHasMissingValues_FilesToAnalyzePath_ReturnsCorrectValue(string folder)
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf\0\FilesToAnalyze.txt");
+        }
+
+        [DataTestMethod]
+        [DataRow("Invalid_DifferentClassName")]
+        [DataRow("Invalid_DifferentNamespace")]
+        [DataRow("Invalid_Xml")]
+        [ExpectedException(typeof(InvalidOperationException), "File SonarProjectConfig.xml could not be parsed.")]
+        public void WhenInvalid_FilesToReturnPath_ReturnNull(string folder)
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().BeNull();
+            sut.ProjectPath.Should().BeNull();
+            sut.FilesToAnalyzePath.Should().BeNull();
+            sut.OutPath.Should().BeNull();
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().BeNull();
+        }
+
+        [DataTestMethod]
+        [DataRow(null)]
+        [DataRow("/foo/bar/do-not-exit")]
+        [DataRow("/foo/bar/x.xml")]
+        public void WhenNoFileFound_ReturnsNull(string folder)
+        {
+            var options = TestHelper.CreateOptions(folder);
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().BeNull();
+            sut.ProjectPath.Should().BeNull();
+            sut.FilesToAnalyzePath.Should().BeNull();
+            sut.OutPath.Should().BeNull();
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().BeNull();
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(InvalidOperationException), "File SonarProjectConfig.xml has been added as an AdditionalFile but does not exist.")]
```

Comment:
Isn't this violation of RSPEC-3431? The `Should().BeNull()` is a smoke and it's not clearly visible where it should fail.
This should do the trick:
```
sut.Invoking(x => x.AnalysisConfigPath).Should().Throw<InvalidOperationException>().WithMessage("File SonarProjectConfig.xml has been added as an AdditionalFile but does not exist.");
```

Reply:
I thought about this but with a test that has 1 line I thought it shouldn't be a problem. Anyway, it's cleaner your way